src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -427,6 +365,75 @@ CHIP_ERROR OTAImageProcessorImpl::InitEncryptedOTA(const CharSpan & key)
     mEncryptedOTAEnabled = true;
     return CHIP_NO_ERROR;
 }
+
+CHIP_ERROR OTAImageProcessorImpl::DecryptStart()
+{
+    VerifyOrReturnError(mEncryptedOTAEnabled, CHIP_ERROR_INCORRECT_STATE);
+
+    const esp_decrypt_cfg_t decryptionConfig = {
+        .rsa_priv_key     = mKey.data(),
+        .rsa_priv_key_len = mKey.size(),
+    };
+
+    mOTADecryptionHandle = esp_encrypted_img_decrypt_start(&decryptionConfig);
+    VerifyOrReturnError(mOTADecryptionHandle, CHIP_ERROR_INCORRECT_STATE);
+    return CHIP_NO_ERROR;
+}
+
+void OTAImageProcessorImpl::DecryptEnd()
```

Comment:
Should this API return an error code? We will not be able to know if the decryption process was successful if we do not have the return code from `esp_encrypted_img_decrypt_end()`

Reply:
`DecryptEnd` is returning an error now and caller is honouring it.